### DISCUSSION TITLE
Application icons in tab title

### DISCUSSION THREAD

**Discussion by georgweiss** [ControlSystemStudio/phoebus#2914] [General]
Currently all application tabs display the app name only. What about also showing the app icon to the left of the name?

**Comment by kasemir** [comment]
Sure, that would be cute.

Beware that the `DockItem`, which is basically a `Tab`, already sets a `Graphic`. The text you see for the name is not the plain Tab text because we need something that's draggable to allow moving the tabs. So the text you see is a `Label` which then becomes the `Graphic` of the `Tab`. To add an icon, the tab Graphic would have to turn into something like `HBox(ImageView, Label)`.

I am generally worried about adding to the scene graph, but this seems reasonable. Worst case, we could replace the 'graphic' with just one image into which we paint the icon and the text, so we prepare that image once, and the scene graph is then one ImageView.
